botorch/acquisition/multi_objective/monte_carlo.py

```diff
@@ -708,9 +730,6 @@ def _set_sampler(
                 self.sampler.base_samples[..., :end_idx, :] = current_base_samples.view(
                     view_shape
                 ).expand(expanded_shape)
-                # update cached subset indices
-                # Note: this also stores self.q = q
-                self._cache_q_subset_indices(q=q)
```

Comment:
Why is this removed?

Reply:
It's redundant since `_compute_qnehvi` has this call. Both `self.q` and `q_subset_indices` are used only after that (or in the `q!=self.q` check).